src/containers/SignInPage.js

```diff
@@ -2,35 +2,27 @@ import React, {PropTypes} from 'react';
 import {browserHistory} from 'react-router';
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
-import * as userActions from '../actions/userActions';
 import * as sessionActions from '../actions/sessionActions';
 import SignInForm from '../components/signin/SignInForm';
 import LandingRightSide from '../components/common/LandingRightSide';
 import cookie from 'react-cookie';
-import userClient from '../client/UsersServerClient';
-import targetClient from '../client/TargetsServerClient';
 
 export const SignInPage = (props) => {
   const user = cookie.load('user');
   if (user) {
-    userClient.signIn(user).then(data => {
-      targetClient.setUserInfo(data.token, data.user_id);
-      props.actions.updateSessionInformation(props.session, "user_id", data.user_id);
-      props.actions.updateSessionInformation(props.session, "user_token", data.token);
-      props.actions.updateSessionInformation(props.session, "isLoggedIn", true);
-      browserHistory.push('/home');
-      return null;
-    }).catch(error => {
-      console.log(error);
-    });
+    props.actions.signIn(user);
   }
+  if (props.session.isLoggedIn == true) {
+    browserHistory.push('/home');
+    return null;
```

Comment:
If i remove the return null statement, i get this error : "Uncaught Error: A valid React element (or null) must be returned. You may have returned undefined, an array or some other invalid object." And the  app would go blank.